nix/keys.nix

```diff
@@ -123,6 +148,16 @@ in
       "things to break."
     )];
 
+    assertions = flip mapAttrsToList config.deployment.keys (key: opts: {
+      assertion = (opts.text == null && opts.filename != null) ||
+                  (opts.text != null && opts.filename == null);
+      message = "Deployment key '${key}' must have either a 'text' or a 'filename' specified.";
+    }) ++        flip mapAttrsToList config.deployment.keys (key: opts: {
+      assertion = !config.deployment.storeKeysOnMachine ||
+                  !opts.filename;
+      message = "On-machine storage of 'filename'-specified deployment keys is not supported. Offending key is: '${key}'.";
```

Comment:
I'd change the wording to: `deployment.storeKeysOnMachine` and `filename`-specified deployment keys is not ...